File: Java/app/src/main/java/com/example/topquiz/controller/MainActivity.java
package com.example.topquiz.controller;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.example.topquiz.R;
import com.example.topquiz.model.User;

public class MainActivity extends AppCompatActivity {

    public static final int GAME_ACTIVITY_REQUEST_CODE = 1;
    public static final String FIRSTNAME = "firstname";
    public static final String SCORE = "score";

    private TextView greetingText;
    private EditText nameInput;
    private Button playbutton;

    private User user;

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (GAME_ACTIVITY_REQUEST_CODE == requestCode && RESULT_OK == resultCode) {
            int score = data.getIntExtra(GameActivity.BUNDLE_EXTRA_SCORE, 0);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        greetingText = findViewById(R.id.activity_main_greeting_txt);
        nameInput = findViewById(R.id.activity_main_name_input);
        playbutton = findViewById(R.id.activity_main_play_btn);

        playbutton.setEnabled(false);

        String firstName = getPreferences(MODE_PRIVATE).getString(FIRSTNAME, null);


        nameInput.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                playbutton.setEnabled(s.toString().length() != 0);

            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        playbutton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                user = new User(nameInput.getText().toString());
                SharedPreferences preferences = getPreferences(MODE_PRIVATE);
                preferences.edit().putString(FIRSTNAME, user.getFirstName()).apply();

                Intent gameActivity = new Intent(MainActivity.this, GameActivity.class);
                startActivityForResult(gameActivity, GAME_ACTIVITY_REQUEST_CODE);

            }
        });


    }
}

File: Kotlin/app/src/main/java/com/example/topquiz_kotlin/controllers/MainActivity.kt
package com.example.topquiz_kotlin.controllers

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.example.topquiz_kotlin.R
import com.example.topquiz_kotlin.models.User

class MainActivity : AppCompatActivity() {

    private lateinit var greetingText: TextView
    private lateinit var nameInput: EditText
    private lateinit var playButton: Button

    private lateinit var user: User

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        greetingText = findViewById(R.id.activity_main_greeting_txt)
        nameInput = findViewById(R.id.activity_main_name_input)
        playButton = findViewById(R.id.activity_main_play_btn)

        playButton.isClickable = false

        nameInput.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable?) {}
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                playButton.isClickable = s?.length != 0
            }
        })

        playButton.setOnClickListener({
            user = User(nameInput.text.toString())
            val preferences: SharedPreferences = getPreferences(Context.MODE_PRIVATE)
            preferences.edit().putString("First name", user.firstName).apply()
            startActivityForResult(Intent(this, GameActivity::class.java), 1)
        })
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (Activity.RESULT_OK == resultCode && requestCode == 1) {
            //
        }
    }


}